### DISCUSSION TITLE
HTTP/1.1과 HTTP/2.0의 차이를 아는대로 알려주세요

### DISCUSSION THREAD

**Discussion by sa02045** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#57] [Interview Question]
HTTP/1.1과 HTTP/2.0의 차이를 아는대로 알려주세요

**Comment by yunjjeongjo** [comment]
### Multiplexed Streams
HTTP 1.0에서 TCP 세션을 맺는 것을 중복해서 수행하는 성능 이슈가 있었고, HTTP 1.1에서 Keep-alive를 통해서 해당 문제를 풀어냈음. 
HTTP 2.0에서는 Multiplexed라는 기술을 도입하는데 1개의 세션으로 여러 개의 요청을 순서 상관없이 Stream으로 받아서 동시다발적으로 처리하고 응답할 수 있게 됨. 이는 특정 요청이 먼저 끝나면 해당 요청에 대해 먼저 응답.
(HTTP 1.1의 HOL 블로킹(Head-Of-Line Blocking)문제를 해결)

### Server Push
기존(HTTP 1.1)에서는 HTML 문서를 요청한 후, 해당 문서에서 필요한 CSS와 Images를 요청했음.
HTTP 2.0부터는 HTML 문서를 요청하면, 클라이언트가 추가로 요청을 하지 않아도 서버가 필요한 리소스를 알아서 보내줌

**Comment by punchdrunkard** [comment]
1.1이랑 2.0의 차이가 latency를 어떻게 해결했느냐의 차이
결론은 성능개선

**Comment by punchdrunkard** [comment]
(추가질문) 3.0과의 차이?
